### PR TITLE
Research doc

### PR DESCRIPTION
**Pull request by FacuDeLorenzo**

### CODE REVIEW

File: docs/research.md
Comment (matzapata) [inline docs/research.md:1]: I think we can do smth like this https://github.com/ethereum-optimism/design-docs/blob/main/protocol/op-contracts-manager-arch.md looks really good, with sections

```
# Porpouse
# Summary
# Problem statement + Context
# Alternatives considered
... 
# Porposed solutions
# Risk and uncertainties 
```

File: docs/research.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Research & Discovery
+
+## Existing similar tools and benefits to our project
+
+### 0xFableOrg/roll-op
+- Rollup customization with config file
```
Comment (matzapata) [inline docs/research.md]: Cons I would add
- It's in python which would drag us to use python for our whole development if we decide to go that way. Thing we don't want as it typically derives in lower quality code.
- Low code quality, a bit messy and poorly structured
- Attached to just one version of the source code and no way to easily adapt for more versions
- Deploys contracts from op-node against advice by the documentation
- No bridge deployment
- Too many dependencies
- Bugs on dependencies checks

Pros I would add:
- Functional, so far only one I could take and run, at least locally.
- Includes example configs and procedures we can replicate
- Includes blockexplorer

File: docs/research.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+### upnodedev/opstack-compose
+- Dev ops solution (docker-compose + aws)
+- Upnode focuses on blockchain infrastructure (more value)
+- Has a block explorer
```
Comment (matzapata) [inline docs/research.md]: I would add as a con:
- doesn't work out of the box
- Same regarding attachment to a version, the build process is hardcoded
- Same regarding deploying contracts from the same release used for building binaries

As pros:
- Does implement some deployment we can inspire ourselves on

File: docs/research.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Decisions and drivers
+
```
Comment (matzapata) [inline docs/research.md]: We decided to spin off our own development of cli and infrastructure tooling to create a more flexible deployment process and solve some of the inconveniences we found while testing existing solutions. Given we couldn't get current solutions to work easily and the time required to replicate them is not that big we aim for our own development to also contemplate the new requirements being a user interface and a scalable deployment

File: docs/research.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+### Developed on Rust
+- Fast, lightweight and compiled
+- Reduces computing costs
+- Increasing popularity
+ 
+## Strategy
+- Core isolated from CLI and App packages enables for Core functionalities testing.
```
Comment (matzapata) [inline docs/research.md]: Also I think to allow for extension and implementation of new multiple clients, in our case the server for the UI, and potentially more

File: docs/research.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- CLI for local self-service
+
+- Deploy with helm on top of `opstack-compose`'s docker-compose
+	- Networking already solved
+	- Cloud provider-agnostic
+	- Provides scalability
+    - Eases deploy process
+
+- Cloud service App: UI + Lambda API implementing Core
+	- Rust lambdas are cheap and scale easily
```
Comment (matzapata) [inline docs/research.md]: I would like to debate more this part, I think we need to better define the UI to say this, if it's a config generator and user tutorial absolutely agree, if we need to do a build process in the back we may need something more powerful...

File: docs/research.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+- Dev ops solution (docker-compose + aws)
+- Upnode focuses on blockchain infrastructure (more value)
+- Has a block explorer
+
+## Decisions and drivers
+
+### Developed on Rust
+- Fast, lightweight and compiled
+- Reduces computing costs
+- Increasing popularity
```
Comment (matzapata) [inline docs/research.md]: Maybe specify why not `go` also, as optimism is in go, we can say like the `op-deployer` can potentially be a go alternative and in this case we're not making a pr to the optimism project itself but living as an individual tool.

File: docs/research.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# Research & Discovery
+
+## Existing similar tools and benefits to our project
+
```
Comment (matzapata) [inline docs/research.md:24]: We can add https://docs.optimism.io/builders/chain-operators/tools/op-deployer and https://github.com/Aymen-Tirchi/op-stack-deployer

File: docs/research.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+We have decided to develop our own CLI and infrastructure tooling to create a more flexible deployment process and address some of the inconveniences we encountered while testing existing solutions. Given the challenges in getting current solutions to work easily and the relatively short time required to replicate them, we aim for our development to also meet new requirements, including a user interface and scalable deployment.
+
+## Decisions and drivers
+
+### General
+
+- A cli that can handle deployments will be developed.
+- A Web UI to quickly and easily spin off projects will be created.
+- We chose Rust for its performance, cost efficiency, and growing popularity, ensuring reliable and scalable solutions.
```
Comment (juandahl) [inline docs/research.md]: I would change growing popularity for sth like `growing community support`.

**Review by juandahl — APPROVED** [review]

**Pull request merged**